File: c2/square_root.c
#include <stdio.h>
#include <math.h>
#define ERRO 1e-12

double square_root(double x) {
  double y = x, anterior;
  do {
    anterior = y;
    y = (y + x / y) / 2;
  } while (fabs(anterior - y) > ERRO);
  return y;
}

int main() {
  double x;
  printf("Entre com o numero:\n");
  scanf("%lf", &x);
  printf("Raiz quadrada: %lf\n", square_root(x));
  return 0;
}

File: c2/square_root.py
ERRO = 1e-12


def square_root(x):
    y = x
    erro_pequeno = False
    while not erro_pequeno:
        anterior = y
        y = (y + x / y) / 2
        if abs(anterior - y) <= ERRO:
            erro_pequeno = True
    return y


print("Entre com o numero:")
x = float(input())
print("Raiz quadrada:", square_root(x))
